diff -uNr cowpatty-4.3/cowpatty.c cowpatty-4.3-hashfix/cowpatty.c
--- cowpatty-4.3/cowpatty.c	2008-03-20 09:49:38.000000000 -0700
+++ cowpatty-4.3-hashfix/cowpatty.c	2008-10-19 23:29:22.000000000 -0700
@@ -202,7 +202,7 @@
 	}
 
 	/* Test that the files specified exist and are greater than 0 bytes */
-	if (!IsBlank(opt->hashfile)) {
+	if (!IsBlank(opt->hashfile) && strncmp(opt->hashfile, "-", 1) != 0) {
 		if (stat(opt->hashfile, &teststat)) {
 			usage("Could not stat hashfile.  Check file path.");
 			exit(-1);
